flax/linen/module.py

```diff
@@ -1217,6 +1218,42 @@ def __call__(self, x):
     scope = core.bind(variables, rngs=rngs, mutable=mutable)
     return self.clone(parent=scope)
 
+  def unbind(self: M) -> Tuple[M, VariableDict]:
+    """Returns an unbound copy of this Module and its variables.
+
+    ``unbind`` is useful for creating a stateless version of a bound Module.
+    A common use case araises when you want extract a submodule defined inside
+    ``setup()``, since ``setup()`` is only called when the Module is bound, a
+    simple way to get hold of the submodule and its corresponding variables is
+    to ``bind`` the parent Module temporarily and then ``unbind`` the desired
+    submodule::
```

Comment:
can / should we add a link to the transfer learning guide?

Reply:
Will be adding a "Using Stateful Modules guide" soon, we can reference that instead when its ready.